### PR TITLE
Dual Contouring LOD

### PR DESCRIPTION
**Pull request by aryasalehi**
### Goal
Creating chunks of terrain with different Level Of Detail that connect seamlessly
### Implementation
For a LOD0 node the previous structure of just Leafs and Internal nodes was fine, but for LOD1 we want to be able to generate the mesh with half as many polygons which are twice the size. 
For LOD2 there would be half as many again and the size would double again, and so on ...
We can enable this by **simplifying the octree**. 
For more info you can visit [this article](http://ngildea.blogspot.com/2015/03/engine-overview.html).
### Seams for different LODs
- The seam generation is the same as before. We gather the seam nodes from our target chunk + the neighbouring chunks 
- The seam nodes may be different sizes so we need to process the input nodes in stages along with the newly constructed parent nodes until all the nodes have the same size
- Finally we construct the seam nodes with the seam octree that has been generated

https://user-images.githubusercontent.com/64514807/166134607-860d8c8b-d99c-401a-b469-a1e20d2c67e0.mp4

### Useful Links
- http://ngildea.blogspot.com/2015/03/engine-overview.html
- http://ngildea.blogspot.com/2014/09/dual-contouring-chunked-terrain.html

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (avaer) [inline .vscode/settings.json:1]: Let's not commit this file.

**Review by avaer — CHANGES_REQUESTED** [review]
Poor editor config hygiene in this PR.

File: .vscode/settings.json (hunk 1 above)
Comment (avaer) [reply .vscode/settings.json:1]: Unless this is intentional?

**Review by avaer — COMMENTED** [review]
Rest of the code seems ok.

File: .vscode/settings.json (hunk 1 above)
Comment (aryasalehi) [reply .vscode/settings.json:1]: No it is committed by mistake, I'll remove it now

**Review by avaer — APPROVED** [review]

**Comment by avaer** [comment]
Needs a rebuild.

**Pull request merged**